spec/rubocop/cop/layout/line_length_spec.rb

```diff
@@ -1235,5 +1235,28 @@ def baz(bar)
         end
       end
     end
+
+    context 'percent array' do
+      context 'with --disable-uncorrectable option' do
+        before { allow(cop).to receive(:disable_uncorrectable?).and_return(true) }
+
+        let(:cop_config) { { 'Max' => 30 } }
+
+        it 'adds an offense and disable it' do
+          expect_offense(<<~RUBY)
+            ARRAY = %i[
+            AAAAAAAAAAAAAAAAAAAA BBBBBBBBBBBBBBBBBBBB].freeze
+                                          ^^^^^^^^^^^^^^^^^^^ Line is too long. [49/30]
+          RUBY
+
+          expect_correction(<<~RUBY)
+            # rubocop:todo Layout/LineLength
+            ARRAY = %i[
+            AAAAAAAAAAAAAAAAAAAA BBBBBBBBBBBBBBBBBBBB].freeze
+            # rubocop:enable Layout/LineLength
+          RUBY
+        end
+      end
+    end
```

Comment:
IMHO, mock testing is naive and should be used as little as possible. Can you replace it with cli/autocorrect_spec.rb instead?